### PR TITLE
DB\Quest: Mage quests

### PR DESCRIPTION
**Pull request by poszer**
Mage Quests ( H/A)

##### CHANGES PROPOSED:

-  Fix quest Investigate the Alchemist Shop - script weren't working.
-  Fix quest Investigate the Blue Recluse - script weren't working.

###### ISSUES ADDRESSED:

Closes 

##### TESTS PERFORMED:

Builded without errors, imported db updates without errors. Tested in game, the quest didn't worked before. Now it's works perfectly.

##### HOW TO TEST THE CHANGES:

1. Create Mage ( A or H )
2. .quest add [1920](https://www.wowhead.com/quest=1920/investigate-the-blue-recluse) (A)
3. quest add [1960](https://www.wowhead.com/quest=1960/investigate-the-alchemist-shop) (H)
4. Teleport to Quest Map location and try to do quest

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

### CODE REVIEW

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -5364,6 +5364,25 @@ void Spell::EffectTransmitted(SpellEffIndex effIndex)
         fz = m_caster->GetPositionZ();
     }
 
+    if (m_spellInfo->Id == 9082) // Investigate the XXX - Mage quest
```
Comment (FrancescoBorzi) [inline src/server/game/Spells/SpellEffects.cpp]: what does the comment mean?

**Comment by Stoabrogga** [comment]
@poszer 
Why this PR? This was already fixed through PR #854. I tested the quest today and it is still working without this PR.

**Comment by FrancescoBorzi** [comment]
@poszer now getting this startup error:

> Script named 'npc_lazy_peon' does not have a script name assigned in database.

File: src/server/game/Spells/SpellEffects.cpp (hunk 1 above)
Comment (poszer) [reply src/server/game/Spells/SpellEffects.cpp]: 2 Quests for Mage's ( Both Factions ) . Quest names are 
1.Investigate the Alchemist Shop
2.Investigate the Blue Recluse. 
That's why XXX ( To skip writing both Quests names )

**Comment by PolluxTroy0** [comment]
Just tried that, and it work.

First try on actual master branch, not working.
Added manually the sql modification to the db, and tried the quest, and it works. Ghosts appear, attack them then used the object in my inventory, and ghost disappear, then i can use and loot the chest on the ground.

I was able to finish the quest ;)

**Comment by Stoabrogga** [comment]
I tested both quests (1920 and 1960) once more and they are working fine **without** this PR. I used the newest commit 987095e2c9da04b6ac9efcd2639a012cc72552d7. Here's also a video showing that          the quest is working:
[2019-01-27 14-45-06.zip](https://github.com/azerothcore/azerothcore-wotlk/files/2800092/2019-01-27.14-45-06.zip)

@poszer @PolluxTroy0 
Perhaps you should check your DB. There were many changes to the structure, perhaps something went wrong while updating.

**Comment by poszer** [comment]
> I tested both quests (1920 and 1960) once more and they are working fine **without** this PR. I used the newest commit [987095e](https://github.com/azerothcore/azerothcore-wotlk/commit/987095e2c9da04b6ac9efcd2639a012cc72552d7). Here's also a video showing that the quest is working:
> [2019-01-27 14-45-06.zip](https://github.com/azerothcore/azerothcore-wotlk/files/2800092/2019-01-27.14-45-06.zip)
> 
> @poszer @PolluxTroy0
> Perhaps you should check your DB. There were many changes to the structure, perhaps something went wrong while updating.

Weird issue. I cant do these quest, installed clean ac with clean db. Also @PolluxTroy0  has tested it, and he confirm the quest doesnt work. Lets wait for next tester ( If the quest doesnt work for him, than your pr didnt solve these quests ). @wetbrownsauce Could you please test it

**Comment by Stoabrogga** [comment]
@poszer 
I also used a new DB for my tests. Could you please describe what the actual error is? What is happening on your side when trying to solve this quest?

**Comment by poszer** [comment]
> @poszer
> I also used a new DB for my tests. Could you please describe what the actual error is? What is happening on your side when trying to solve this quest?

When i come near that rifts, and using item to attack me, i droo them to 1% hp and while using item on them, the object is spawned but i cant loot it, bcoz that object disappear instantly

**Comment by Stoabrogga** [comment]
@poszer 
I think you are doing it wrong:
1. Use "Cantation of Manifestation" -> Rift Spawns appear
2. Attack the Rift Spawn until it is stunned
3. **Now** use "Chest of Containment Coffers" -> "Containment Coffer" appears under the stunned Rift Spawn
4. Click on the "Containment Coffer" -> Rift Spawn is sucked into the coffer resulting in a "Filled Containment Coffer"
5. Click on the "Filled Containment Coffer" to pick it up

**Comment by poszer** [comment]
> @poszer
> I think you are doing it wrong:
> 
> 1. Use "Cantation of Manifestation" -> Rift Spawns appear
> 2. Attack the Rift Spawn until it is stunned
> 3. **Now** use "Chest of Containment Coffers" -> "Containment Coffer" appears under the stunned Rift Spawn
> 4. Click on the "Containment Coffer" -> Rift Spawn is sucked into the coffer resulting in a "Filled Containment Coffer"
> 5. Click on the "Filled Containment Coffer" to pick it up

Thats excatly what i did, but object disappear instantly

**Comment by poszer** [comment]
CLOSING THIS since @Stoabrogga say its fixed with hes PR. If needed, when someone open issue about it, ill reopen this.

**Comment by Stoabrogga** [comment]
Please leave this PR open. I really want to sort this issue out. A quest playing out differently using the same core and DB is worrying.
@wetbrownsauce
Could you please take a look at this and test the quest before applying this PR?

**Comment by PolluxTroy0** [comment]
I've done the exact same thing as Poszer. On fresh AC install, not working, same issue as Poszer, item on the ground disappear. Updated my db, on a new freshly build, and here it work.

**Comment by poszer** [comment]
> Tested without PR.
> 
> * Fresh compile of AC core using [987095e](https://github.com/azerothcore/azerothcore-wotlk/commit/987095e2c9da04b6ac9efcd2639a012cc72552d7)
> * Re-created DB from scratch. `version_db_auth` on update 2019_01_05_00 | `version_db_characters` on update 2019_01_06_00 | `version_db_world` on update 2019_01_26_00
> * Tested quest in-game WITHOUT this PR.
> * Everything works as intended.
> 
> Video of quest without PR: [gfycat webm](https://giant.gfycat.com/RecentCharmingIguanodon.webm)
> 
> ###### sorry it cuts off at the end, but I looted the second object same as the first and it worked properly.

Thanks for testing. Anyway on your video its not correct. While using that item to show of rifts creatures, there should be emote from them, and also they should attack you.

**Comment by wetbrownsauce** [comment]
I was only testing if the object disappears instantly like you described.

**Comment by poszer** [comment]
> I was only testing if the object disappears instantly like you described.

Yeah, its okay. Thank you!

**Comment by poszer** [comment]
Dunno why it doesn't wokr on my core. Posible something was incorrect in DB, but don't know why.
Just tested on clean latest version, and it's working. I'm closing this. Thanks @Stoabrogga  for pointing more on this.

**Comment by poszer** [comment]
Also could you tell me how you compile core, relwithdbinfo, release or debug mode?

**Comment by Stoabrogga** [comment]
release

**Comment by BarbzYHOOL** [comment]
@PolluxTroy0 debug or relwithdb I suppose?

**Comment by Stoabrogga** [comment]
Tested with Debug and RelWithDebInfo, the quests did work as they did with Release.

**Comment by wetbrownsauce** [comment]
Every test I do is in debug mode, including this one

**Pull request closed without merging**